server/src/main/java/org/elasticsearch/index/mapper/Mapper.java

```diff
@@ -53,6 +56,21 @@ public Settings indexSettings() {
         public Version indexCreatedVersion() {
             return Version.indexCreated(indexSettings);
         }
+
+        public void putPositionInfo(Mapper mapper, Integer position) {
+            if (position == null) {
+                this.columnPositionResolver.addColumnToReposition(mapper.name(),
+                                                                  mapper,
+                                                                  (m, p) -> m.position = p,
+                                                                  contentPath.currentDepth());
+            } else {
+                this.columnPositionResolver.updateMaxColumnPosition(position);
+            }
+        }
+
+        public ColumnPositionResolver<Mapper> getColumnPositionResolver() {
```

Comment:
Is it possible to keep the resolver encapsulated?

Reply:
Since `RootObjectMapper` has a `ColumnPositionResolver` field, it should come from `Mapper.BuilderContext` in the end. Should I try to remove the `ColumnPositionResolver` field from `RootObjectMapper`?